Automattic/wp-calypso
# Apps: Add apps dir and move o2-blocks there

Adds an apps dir for packages that aren't intended for code sharing but may benefit from inclusion in the monorepo.

* Add `apps/`
* Add `apps/*` to `learna.config`
* Move `o2-blocks` to `apps/`
* Change `o2-blocks` compilation script to `build`, setting a precedent that _packages `prepare`_ and _apps `build`_.


In the future, Calypso client, server, desktop and more are good candidates to move into `apps/`.

This allows better alignment with lerna's expectations:

> The packages config in `lerna.json` is a list of globs that match directories containing a `package.json`, which is how lerna recognizes "leaf" packages (vs the "root" `package.json`, which is intended to manage the dev dependencies and scripts for the entire repo).

https://github.com/lerna/lerna#lernajson

## To do

- [ ] [Update docs](https://github.com/Automattic/wp-calypso/pull/32058#issuecomment-480185024)

## Testing instructions

* `npm run postinstall` and `npm run build-packages` should build shared packages in `packages/*`, not `o2-blocks`.
* `npx lerna run build --scope='*/o2-blocks'` should build the `o2-blocks` package. Check its output at `apps/o2-blocks/dist`.
